src/mlpack/tests/main_tests/fastmks_test.cpp

```diff
@@ -0,0 +1,658 @@
+  SetInputParam("scale", 4.0);
+
+mlpackMain();
+
+  CheckMatricesNotEqual(hyptanKernel,
+      CLI::GetParam<arma::mat>("kernels"));
+}
+
+/** Ensure that bandwidth affects the final result of Gaussian, Epanechnikov,
+ *  and triangular kernel
+ */
+
+BOOST_AUTO_TEST_CASE(FastMKSBandwidthTest)
+{
+  // 100 points in 3 dimensions.
+  arma::mat referenceData(3, 100, arma::fill::randu);
+
+  // Random input, some k <= number of reference points.
+  SetInputParam("reference", referenceData);
+  SetInputParam("k", (int) 10);
+  SetInputParam("kernel", (string)"gaussian");
+  SetInputParam("bandwidth", 1.0); // Default value.
+
+  mlpackMain();
+
+  arma::mat gaussianKernel;
+  gaussianKernel = std::move(CLI::GetParam<arma::mat>("kernels"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["bandwidth"].wasPassed = false;
+
+  SetInputParam("reference", referenceData);
+  SetInputParam("bandwidth", 4.0);
+
+  mlpackMain();
+  CheckMatricesNotEqual(gaussianKernel,
+      CLI::GetParam<arma::mat>("kernels"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["bandwidth"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["kernel"].wasPassed = false;
+
+  // Random input, some k <= number of reference points.
+  SetInputParam("reference", referenceData);
+  SetInputParam("kernel", (string)"epanechnikov");
+  SetInputParam("bandwidth", 1.0); // Default value.
+
+  mlpackMain();
+
+  arma::mat epanKernel;
+  epanKernel = std::move(CLI::GetParam<arma::mat>("kernels"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["bandwidth"].wasPassed = false;
+
+  SetInputParam("reference", referenceData);
+  SetInputParam("bandwidth", 4.0);
+
+  mlpackMain();
+  CheckMatricesNotEqual(epanKernel,
+       CLI::GetParam<arma::mat>("kernels"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["bandwidth"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["kernel"].wasPassed = false;
+
+  // Random input, some k <= number of reference points.
+  SetInputParam("reference", referenceData);
+  SetInputParam("kernel", (string)"triangular");
+  SetInputParam("bandwidth", 1.0); // Default value.
+
+  mlpackMain();
+
+  arma::mat triKernel;
+  triKernel = std::move(CLI::GetParam<arma::mat>("kernels"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["bandwidth"].wasPassed = false;
+
+  SetInputParam("reference", referenceData);
+  SetInputParam("bandwidth", 4.0);
+
+  mlpackMain();
+
+  CheckMatricesNotEqual(triKernel,
+      CLI::GetParam<arma::mat>("kernels"));
+}
```

Comment:
We should add a blank line between the end of this method and `BOOST_AUTO_TEST_SUITE_END()`. :+1: